src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -561,13 +561,19 @@ public void UpdatePackageSource(PackageSource source, bool updateCredentials, bo
         {
             if (string.Equals(newSource.Name, existingSource.Name, StringComparison.OrdinalIgnoreCase))
             {
-                if ((!string.Equals(newSource.Source, existingSource.Source, StringComparison.OrdinalIgnoreCase) ||
-                    newSource.ProtocolVersion != existingSource.ProtocolVersion) && newSource.IsPersistable)
+                if (!string.Equals(newSource.Source, existingSource.Source, StringComparison.OrdinalIgnoreCase) && newSource.IsPersistable)
                 {
                     Settings.AddOrUpdate(ConfigurationConstants.PackageSources, newSource.AsSourceItem());
                     isDirty = true;
                 }
 
+                if ((newSource.ProtocolVersion != existingSource.ProtocolVersion) && newSource.IsPersistable)
+                {
+                    Settings.Remove(ConfigurationConstants.PackageSources, existingSource.AsSourceItem());
+                    Settings.AddOrUpdate(ConfigurationConstants.PackageSources, newSource.AsSourceItem());
```

Comment:
@dotnokato 
PR looks great. Just 1 tiny improvement. 
Here we're writing to config file 2nd time if protocol version is different after writing same info in 566 if new and old sources urls are different and protocols are different too.
Could you improve to make it just 1 write to disc?

Reply:
Good catch!
I've pushed updated code for the above.

Reply:
Thank you. LGTM, I requeued the CI build and waiting for result.

Reply:
Sorry, for looking back at code changes tiny improvement added 30 lines of code. Considering this code path is not hot code path previous implementation is good enough for me, I prefer clarity over 30 lines of code.

Reply:
I don't feel comfortable with adding another 30 lines of code, just sake of keeping order in config, if user is really concerned about ordering, then can be fixed manually. For me keeping ordering is another problem. I prefer clarity of code over some optimizations. Considering this is not really hot path, I'm ok with reverting back to 2 write to disc.
Sorry, I edited my comments 2-3 times.

Reply:
I've pushed new changes (see my comment [here](https://github.com/NuGet/NuGet.Client/pull/5002#issuecomment-1679707317)) reverting this bit as it is no longer needed.